src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1368,7 +1368,10 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			this._pressAnyKeyToCloseListener = undefined;
 		}
 
-		this._exitReason = reason || TerminalExitReason.Unknown;
+		if (typeof this._exitReason === 'undefined') {
```

Comment:
This is needed because https://github.com/microsoft/vscode/pull/152791 got reverted and `dispose` can be called multiple times and we don't want to override the initial `_exitReason`